alternative-1-iterate-generic.Rmd

```diff
@@ -78,81 +82,109 @@ iterate.SquaresSequence <- function(x, state = NULL) {
     state <- 1L
   if (state > x$to)
     return(NULL)
-  list(value = (x$from - 1L + state) ^ 2, 
-       state = state + 1L)
+  list(
+    value = (x$from - 1L + state) ^ 2, 
+    state = state + 1L
+  )
 }
 
-for (x in SquaresSequence(1, 10))
+for (x in SquaresSequence(5, 10))
   print(x)
 ```
 
-Reticulate support might look like:
+## Sequence of unknown length
 
 ```{r}
-iterate.python.builtin.object <- function(x, state = NULL) {
-  if(is.null(state))
-    state <- reticulate::as_iterator(x)
+SampleSequence <- function(max) {
```

Comment:
I think it's worth including a standard example of a sequence with unknown length. This isn't a very interesting example, so better ideas are welcome.